x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderServiceTests.java

```diff
@@ -665,14 +668,20 @@ public void testScaleUp_withWaitingJobsAndRoomInNodes() {
                 analytics,
                 List.of(),
                 null,
-                NativeMemoryCapacity.ZERO,
+                currentScale,
```

Comment:
Several of the tests in this file weren't setting `currentScale` appropriately for the test scenario, and were then asserting results that didn't make sense given how things work in production. I have tried to make all the `testScaleUp_blah` tests in this file call `service.checkForScaleUp()` with coherent arguments so that we will be exercising that method in the same way as production in the future.